bonsai/Modules/BudgetDetails/BudgetViewModelAssembler.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  BudgetViewModelAssembler.swift
+//  bonsai
+//
+//  Created by antuan.khoanh on 17/07/2022.
+//
+
+import Foundation
+import CoreData
+
+struct BudgetViewModelAssembler: Assembler {
+   private var transactions: [Transaction] = []
+
+   init(mainContext: NSManagedObjectContext = DataController.sharedInstance.container.viewContext) {
+      let fetchRequest: NSFetchRequest<Transaction> = Transaction.fetchRequest()
+      do {
+         let transactions = try mainContext.fetch(fetchRequest)
+         self.transactions = transactions
+      } catch {
+         
```

Comment:
I suggest to at least add assert in catch block

Reply:
updated +